contracts/yellow/IVault.sol

```diff
@@ -0,0 +1,113 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.16;
+
+/**
+ * @notice IVault is the interface to implement custody.
+ */
+interface IVault {
+    // ======================
+    // Structs
+    // ======================
+    struct Allocation {
+        address asset;
+        uint256 amount;
+    }
+
+    // ======================
+    // Functions
+    // ======================
+    /**
+     * @notice Get last ledger id (deposits and withdrawals id).
+     * @return uint256 Ledger id.
+     */
+    function getLastId() external view returns (uint256);
+
+    /**
+     * @notice Set the address derived from the broker's new public key. Emits `BrokerKeyDerivedAddressSet` event.
+     * @dev Supplied payload must be signed by broker's current public key.
+     * @param encodedAddress Encoded address to set.
+     * @param signature Encoded address signed by broker's current public key.
+     */
+    function setBrokerKeyDerivedAddress(bytes calldata encodedAddress, bytes calldata signature) external;
```

Comment:
why the address is not a address type?

Reply:
To recover the signer of a message, we use OpenZeppelin's `ECDSA` library, including `toEthSignedMessageHash` method, which receives hashed data. In solidity we can only hash data of type `bytes`. Of course it is possible to 'convert' (but actually encode) `address` to `bytes` by calling `abi.encode(myAddress)`, but that would also require encoded address signature supplied. Therefore the function receive encoded address and on the server side we will need to encode address too.